src/connection_manager.rs

```diff
@@ -75,6 +75,9 @@ impl std::default::Default for OperationRetryOptions {
 pub struct TlsOptions {
     /// contains a list of PEM encoded certificates
     pub certificate_chain: Option<Vec<u8>>,
+
+    /// Allow insecure TLS connection if set to true
+    pub allow_insecure_connection: bool
```

Comment:
can we have one more flag like `validate_hostname`?

Reply:
Could you explain a bit more?

Reply:
@fantapsody i noticed that you set both `danger_accept_invalid_hostnames` and `danger_accept_invalid_certs` according to `allow_insecure_connection`. My suggestion is to use 2 options, like `validate_hostname` and `allow_insecure_connection`, so that it will act like other pulsar client.

Reply:
Thanks, I see, there is a configuration `isTlsHostnameVerificationEnable` in the Java pulsar cluster.